### DISCUSSION TITLE
Is it possible to lazyload overseer and also be part of the lualine?

### DISCUSSION THREAD

**Discussion by dhruvinsh** [stevearc/overseer.nvim#353] [Q&A] [answered]
Hi,
as the title says, is it possible to lazyload overseer by command or by keys and also be part of the lualine?

**Comment by stevearc** [comment]
The component could probably be refactored to support lazy-loading, but it would be tricky because of the rigid expected structure. Also, given that the component is defined in the overseer plugin, I'm not sure that there would be any way to lazy load overseer while referencing that component.

**Reply by dhruvinsh** [reply, marked as answer]
thank you for looking into it, but solved the brain teaser puzzle.

```lua 
-- overseer.lua file
return {
  "stevearc/overseer.nvim",
  cmd = { "OverseerClose", "OverseerOpen", "OverseerRun", "OverseerToggle" },
  config = function(_, opts)
    local overseer = require("overseer")
    overseer.setup(opts)
    do
      local ok, lualine = pcall(require, "lualine")
      if not ok then
        return
      end
      local lualine_cfg = lualine.get_config()
      table.insert(lualine_cfg.sections.lualine_x, 1, "overseer")
      lualine.setup(lualine_cfg)
    end
  end,
}

-- lualine.lua file
return {
  "nvim-lualine/lualine.nvim",
  opts = {},
}
```
with index based table insert I still keep the `overseer` at the first and any other component as is.
